modules/javafx.graphics/src/main/java/com/sun/javafx/font/WindowsFontMap.java

```diff
@@ -25,103 +25,97 @@
 
 package com.sun.javafx.font;
 
-import java.util.HashMap;
+import java.util.Map;
 
 class WindowsFontMap {
 
-    static class FamilyDescription {
-        public String familyName;
-        public String plainFullName;
-        public String boldFullName;
-        public String italicFullName;
-        public String boldItalicFullName;
-        public String plainFileName;
-        public String boldFileName;
-        public String italicFileName;
-        public String boldItalicFileName;
+    private static class FamilyDescription {
```

Comment:
Isn't that basically a `record`  in disguise?
Letting this class be a record would remove a lot of lines, especially down below.
Also, it would come with the benefit of not having to worry about whether these values have been changed or not as they would then always be final.
To me at least it does not look as if variables of this class should be mutable after having been created.

Reply:
We still build JavaFX with `--release 11` so applications using JavaFX can run on JDK 11 or later. At some point we will bump the minimum version of Java that is required, but until then we cannot use records or any other feature that isn't in Java 11.

Reply:
That is indeed a valid argument as to why this isn't a record.
Obviously, I did not think about that while reviewing.

Reply:
@delvh If you are interested, I already have a branch with records replacements for when we are able to use them https://github.com/nlisker/jfx/tree/Records. It hasn't been updated in a while so it doesn't include everything on my list.